gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowConfigResourceLocalHandler.java

```diff
@@ -108,9 +108,32 @@ public FlowConfig getFlowConfig(FlowId flowId) throws FlowConfigLoggedException
    */
   public CreateResponse createFlowConfig(FlowConfig flowConfig, boolean triggerListener) throws FlowConfigLoggedException {
     log.info("[GAAS-REST] Create called with flowGroup " + flowConfig.getId().getFlowGroup() + " flowName " + flowConfig.getId().getFlowName());
-    FlowSpec flowSpec = createFlowSpecForConfig(flowConfig);
-    this.flowCatalog.put(flowSpec, triggerListener);
-    return new CreateResponse(new ComplexResourceKey<>(flowConfig.getId(), new EmptyRecord()), HttpStatus.S_201_CREATED);
+    FlowSpec flowSpec;
+
+    if (!isValidFlowConfig(flowConfig)) {
+      throw new FlowConfigLoggedException(HttpStatus.S_400_BAD_REQUEST,
+          "FlowExecutionId cannot be passed for a scheduled job", null);
+    } else if (flowConfig.hasSchedule() && flowConfig.getSchedule().isRunImmediately() &&
+        flowConfig.getProperties().containsKey(ConfigurationKeys.FLOW_EXECUTION_ID_KEY)) {
+      flowSpec = createFlowSpecForConfig(flowConfig, flowConfig.getProperties().get(ConfigurationKeys.FLOW_EXECUTION_ID_KEY));
+      this.flowCatalog.put(flowSpec, triggerListener);
+      FlowStatusId flowStatusId = new FlowStatusId()
+          .setFlowName(flowSpec.getConfigAsProperties().getProperty(ConfigurationKeys.FLOW_NAME_KEY))
+          .setFlowGroup(flowSpec.getConfigAsProperties().getProperty(ConfigurationKeys.FLOW_GROUP_KEY))
+          .setFlowExecutionId(Long.parseLong(flowSpec.getConfigAsProperties().getProperty(ConfigurationKeys.FLOW_EXECUTION_ID_KEY)));
+      return new CreateResponse(new ComplexResourceKey<>(flowConfig.getId(), flowStatusId), HttpStatus.S_201_CREATED);
+    } else {
+      flowSpec = createFlowSpecForConfig(flowConfig);
+      this.flowCatalog.put(flowSpec, triggerListener);
+      FlowStatusId flowStatusId = new FlowStatusId().setFlowGroup(flowSpec.getConfigAsProperties().getProperty(ConfigurationKeys.FLOW_GROUP_KEY))
+          .setFlowName(flowSpec.getConfigAsProperties().getProperty(ConfigurationKeys.FLOW_NAME_KEY));
+      return new CreateResponse(new ComplexResourceKey<>(flowConfig.getId(), flowStatusId), HttpStatus.S_202_ACCEPTED);
+    }
+  }
+
+  private static boolean isValidFlowConfig(FlowConfig flowConfig) {
+    return !flowConfig.hasSchedule() || StringUtils.isEmpty(flowConfig.getSchedule().getCronSchedule()) ||
+        !flowConfig.getProperties().containsKey(ConfigurationKeys.FLOW_EXECUTION_ID_KEY);
   }
```

Comment:
The definition of isValidFlowConfig() is a little confusing. It seems like it should capture all the conditions under which you expect to see the FLOW_EXECUTION_ID in the flowConfig. 
1. When the flowConfig specifies a "run-once" flow, OR
2. When it is a scheduled flow, but has "run-immediately" set to true.

Also, something to think about: should the validation check be moved to createFlowSpecForConfig() method?